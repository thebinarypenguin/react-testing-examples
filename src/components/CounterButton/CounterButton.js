import React from 'react';

class CounterButton extends React.Component {

  constructor(props) {
    super(props);

    this.state = {};

    // Set initial value for count
    if (Number.isInteger(props.initialCount)) {
      this.state.count = props.initialCount;
    } else {
      this.state.count = 0;
    }

    this.increment = this.increment.bind(this);
  }

  increment() {
    this.setState({ count: this.state.count + 1 });
  }

  render () {

    return (
      <button onClick={this.increment}>
        I have been clicked {this.state.count} {this.state.count === 1 ? 'time' : 'times' }
      </button>
    )
  }
}

export default CounterButton;
