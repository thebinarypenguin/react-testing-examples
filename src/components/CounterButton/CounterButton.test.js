import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CounterButton from './CounterButton';

/*
 * Smoke Test
 * When we render the component does it cause any errors?
 */
it('renders without crashing', () => {

  const div = document.createElement('div');

  ReactDOM.render(<CounterButton />, div);

  ReactDOM.unmountComponentAtNode(div);
});

/*
 * Snapshot Test
 * Does the component render the correct UI
 */
it('renders the UI as expected', () => {

  // Render the component and convert it's output into a JSON format called a snapshot.
  // These snapshot files are created in the directory ./__snapshots__
  const tree = renderer.create(<CounterButton />).toJSON();

  // Assert the current snapshot matches the correct snapshot.
  // The very first time you run this test you won't have anything to compare to
  expect(tree).toMatchSnapshot();
});
