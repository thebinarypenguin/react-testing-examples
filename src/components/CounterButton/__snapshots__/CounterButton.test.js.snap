// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<button
  onClick={[Function]}
>
  I have been clicked 
  0
   
  times
</button>
`;
